xbmc/playlists/SmartPlayList.cpp

```diff
@@ -815,6 +824,8 @@ std::string CSmartPlaylistRule::FormatWhereClause(const std::string &negate, con
       query = GetField(m_field, strType) + " is NULL or " + GetField(m_field, strType) + parameter;
     else if (m_field == FieldSource)
       query = negate + " EXISTS (SELECT 1 FROM album_source, source WHERE album_source.idAlbum = " + GetField(FieldId, strType) + " AND album_source.idSource = source.idSource AND source.strName" + parameter + ")";
+    else if (m_field == FieldDiscTitle)
+      query = negate + " EXISTS (SELECT 1 FROM album JOIN song ON song.idAlbum = album.idAlbum WHERE album.idAlbum = " + GetField(FieldId, strType) + " AND song.strDiscSubtitle" + parameter + ")";
```

Comment:
No need for JOIN from album. Replace with
`query = negate + " EXISTS (SELECT 1 FROM song WHERE song.idAlbum = " + GetField(FieldId, strType) + " AND song.strDiscSubtitle" + parameter + ")";`